### ISSUE TITLE
Nested objects within auth not working.

### ISSUE THREAD

**Issue by pthrasher** [goldibex/targaryen#75]
I've created a simple test that fails in targaryen, but passes on firebase simulator.

###### Test:

```json
{
  "root": {
    "chats": {
      "chat-01": {
        "orgID": "org-01"
      }
    }
  },

  "users": {
    "user01": {
      "uid": "user-01",
      "orgs": {
        "org-01": true
      }
    }
  },

  "tests": {
    "chats/chat-01": {
      "canRead": [ "user01" ]
    }
  }
}
```

###### Rules:

```json
{
  "rules": {
    "chats": {
      "$chatID": {
        ".read": "root.child('chats').child($chatID).child('orgID').val() != null && auth.orgs != null && auth.orgs[root.child('chats').child($chatID).child('orgID').val()] == true"
      }
    }
  }
}
```

For easier reading, here's the rule pretty-printed:

```
root.child('chats').child($chatID).child('orgID').val() != null
&&
auth.orgs != null
&&
auth.orgs[ root.child('chats').child($chatID).child('orgID').val() ] == true
```

**Comment by dinoboff** [comment]
@pthrasher thanks.

Fixed on master. Will be released today in v2.3.0.

**Comment by dinoboff** [comment]
@pthrasher

ps: sorry about the PR.
pps: v2.3.0 is released

**Comment by simenbrekken** [comment]
I might be misunderstanding but I'm suddenly getting failed tests after upgrading to 2.3.0 from 2.2.0.

The following passed in 2.2.0 but no longer passes in 2.3.0:

Rule: `root.child('projects').child($projectId).child('users').child(auth.uid).val() != null`

Context:
```
$projectId = 'palace'
auth = null
root = { projects: { palace: { users: { frank: { role: 'user } } } } }
```

The evaluated expression is then: `root.projects.palace.users[null].val() != null` which should fail.

**Comment by dinoboff** [comment]
It's the same issue as #86 in the next branch. it's related #65. I reopened the issue.

**Comment by dinoboff** [comment]
@simenbrekken It should be fixed v2.3.2
